### DISCUSSION TITLE
Why everthing should be Double ?

### DISCUSSION THREAD

**Discussion by X901** [skiptools/skip#76] [Q&A] [answered]
Hi, 
I'm working in my first project with Skip
I notice something annoying that I need to convert everthing to Double

add padding ? it must be Double 
add spacing ? it must be Double

in iOS side it working fine
but when run the app it won't run until it converted to Double

isn't it better to convert it with transpiler ? 
so on iOS it's Int but on Android it's Double ?

I don't think there's anyone write Double for Padding and Spacing in SwiftUI

**Comment by marcprux** [comment, marked as answer]
Skip requires you to be explicit with numeric types. This is discussed at: https://skip.tools/docs/swiftsupport/#swift-types

Swift's `Int` (64-bit) type is interpreted as a Kotlin `Int` (32-bit), and doubles literals need to either be specified using a decimal point (e.g., `1.0`, rather than just `1`). Every other numeric type needs to be specified explicitly, such as `UInt64(123)` or `Float(3.14159)`..

We are aware that this can be cumbersome, but if you follow the guidance to always specify double literals with a decimal, and to fully-declare all other numeric types, it becomes second nature.
